quizzes/quiz6.md

```diff
@@ -0,0 +1,313 @@
+
+```js
+divElement.textContent = "God day to you sir";
+divElement.style.backgroundColor = "red";
+
+inputElement.type = "number";
+inputElement.onchange = function() {
+    console.log("user entered a new value");
+}
+```
+
+So in short, whenever you need to add new content to your dom, you start with creating a new object. Initially, every object you create using this method will be invisible to the user. You can add this element anywhere in your dom hierarchy. Because of this variability, you need to specify where you want this object to be added. For learning how to do that keep on reading...
+
+---
+
+**appendChild**
+
+So you created an object and you don't see it on the screen. Maybe you have forgotten to add it to the dom, by telling JavaScript where to insert your element. To fix that, use `appendChild`. When using `appendChild`, keep in mind that your object needs to connect to one visible element. This could be in one of two ways:
+
+1. Use the `document.body` as your insertion place.
+2. Find another element which you have inserted before and be the child of that element.
+
+Example for the usage of the first way:
+
+```js
+document.appendChild(productElement);
+```
+
+For the second way:
+
+```js
+productElement.appendChild(inputElement);
+```
+
+Here we are assuming that `productElement` was inserted to the dom previously.
+
+---
+
+**render function**
+
+What is a render function, a question asked by many students before you. According to wikipedia, rendering is generating an image from a model by means of computer programs. This definition fits the purpose of our `render` function really well.
+
+In our web store application, we created objects, which hold the important data for our products, such as the amount, price, description. This data on its own is not displayable in the browser. We can not submit our data to browser and say display them. We need to convert this data into visual components that the browser understands, which the user can interact with. These components are html elements. In our application, we created and displayed the html elements inside this function.
+
+Every time we are faced with conversion from data into html elements, this operation can be called **"rendering"**. This function could be named differently, but it is customary to call it render.
+
+---
+
+**onchange property**
+
+Let's remember the context where we used this property and generalize the idea behind this concept, which is event handling in JavaScript.
+
+```js
+...
+let inputElement = document.createElement("input");
+inputElement.type = "number";
+...
+inputElement.onchange = function() {
+    console.log(inputElement.value);
+    ...
+}
+```
+
+Previously, we saw how to respond to `click` events in JavaScript, by defining a function in js file and assigning this function in the `onclick` attribute of a `button` element. This was static way of registering an event handler, because we did it in the html file.
```

Comment:
```suggestion
Previously, we saw how to respond to the `click` events in JavaScript, by defining a function in the JavaScript file and assigning this function in the `onclick` attribute of a `button` element in HTML. This was the static way of registering an event handler, because we did it in the HTML file.
```